src/crimson/os/seastore/onode_manager/staged-fltree/stages/key_layout.h

```diff
@@ -140,9 +140,9 @@ struct string_key_view_t {
   // presumably the maximum string length is 2KiB
   using string_size_t = uint16_t;
   static constexpr auto MAX = std::numeric_limits<string_size_t>::max();
-  static constexpr auto MIN = string_size_t(0u);
-  static auto is_valid_size(size_t size) {
-    return (size > MIN && size < MAX);
+  static constexpr auto MIN = MAX - 1;
+  static bool is_valid_size(size_t size) {
+    return size < MIN;
```

Comment:
I don't really understand the role of MIN here.

Reply:
You can assume `MIN` and `MAX` are invalid/impossible values now. They are currently reserved to implement ns/oid deduplication across onode tree nodes.

Reply:
Please also refer to the comments for `struct string_key_view_t`.

Reply:
I mostly object to the fact that < MIN is "valid".

Reply:
OK, I'll add a `UPPER_BOUND` to be `MAX - 2`

Reply:
see https://github.com/ceph/ceph/pull/39333/commits/54f87106649aafa1c6a0cf77e72eb04df3a5a41c